### ISSUE TITLE
Coffee machine display not working

### ISSUE THREAD

**Issue by flashmandv** [chris7topher/SmartPhilips2200#10]
Hey Christopher,

Thank you for the awesome library. I'm trying to make my coffee machine smart (Phillips EP3246).
Using your code I'm able to get machine status successfully and I'm able to send commands, but the display refuses to work.
As I see the Tx pin is router through the ESP8266 but it seems my ESP is not able to route the communication to the display.
Not sure why..maybe I'm using the wrong pin?
I'm using NodeMCU dev board (so no voltage converter is needed).

Since my machine is 3200 series - I'm using this modified version of your code: [https://github.com/mkorenko/esp-phillips-3200](https://github.com/mkorenko/esp-phillips-3200)
I've seen in the code the pinouts are different depending on the board used. Not sure if that is the problem
`#if defined(ESP8266)
DisplayOutSerial.begin(115200);
#else
DisplayOutSerial.begin(115200, SERIAL_8N1, RX2_PIN, TX2_PIN);
#endif` 

Here is my wiring and my Arduino ide project:
<img width="1749" alt="wiring" src="https://github.com/mkorenko/esp-phillips-3200/assets/2466808/b01b4440-a1cd-4a54-a0f9-b152d27422f3">

<img width="500" alt="arduino-ide" src="https://github.com/mkorenko/esp-phillips-3200/assets/2466808/a36f8c92-92e7-405b-bba1-d878d1400399">

Could my code for setting LED pinout (used for testing) be a problem? `pinMode(2, OUTPUT);`
Or maybe incorrect board I've set in Arduino IDE? 

Any help is welcome as I'm stuck.
Big thank you!

**Comment by flashmandv** [comment]
It turned out that the display works fine if I power on the machine from its button. Then it works and it shows what I command sent.
But If I power on the machine from the  ESP, then the display does not turn on at all..no matter what commands I sent.
Any idea on what can it be? Did I messed up with the transistor?

Thanks

**Comment by chris7topher** [comment]
Hi Flashmandv,

in my circuit, I have a transistor that I use to briefly cut off power to the display. This interruption in power is what triggers the display to start. 

BR
Christopher

**Comment by flashmandv** [comment]
Yes, thanks for the answer. I was using the transistor too..but it didn't work until I cut the second GND pin (pin 3 on the machine cable). Now most things are working but I get like ghost randomly starting/stopping start brewing commands (maybe noise..not sure). I just asked for help in this thread of your repo: [https://github.com/chris7topher/SmartPhilips2200/issues/9#issuecomment-1875897752](https://github.com/chris7topher/SmartPhilips2200/issues/9#issuecomment-1875897752) 

I'll be very thankful if you give me some advice on that ghost auto start commands the machine is doing

**Comment by flashmandv** [comment]
The autostart problem turned out to be a defective ESP8266 (probably)...or maybe bad connection (I did not used pullup resistors..mayne I should have tried).

I fixed it by replacing the ESP8266 with ESP32. Added some wifi web interface (server) and tried to add presets (but failed as the water/strangth level did not work due to the machine not updating its status in real time..despite trying with delays).
Nevermind, the custom coffee settings work great. I can turn on/off, start any type of coffee with setting it setting by setting and is all good.
Closing this for now. If anyone wants to contribute - here is my fork for Phillips 3246 [https://github.com/flashmandv/phillips-3246-smart](https://github.com/flashmandv/phillips-3246-smart)
